### PR TITLE
[CALCITE-3452] Support Exchange and Sample in RelMdNodeTypes

### PR DESCRIPTION
**Pull request by xy2953396112**
compensation Exchange node type in RelMdNodeTypes, Same as the [CALCITE-3450].

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java [hunk 1]
```diff
@@ -2360,6 +2361,27 @@ private void checkNodeTypeCount(String sql, Map<Class<? extends RelNode>, Intege
...
+    final RelMetadataQuery mq = rel.getCluster().getMetadataQuery();
+    final Multimap<Class<? extends RelNode>, RelNode> result = mq.getNodeTypes(exchange);
+    assertThat(result, notNullValue());
+    final Map<Class<? extends RelNode>, Integer> resultCount = new HashMap<>();
+    for (Entry<Class<? extends RelNode>, Collection<RelNode>> e : result.asMap().entrySet()) {
+      resultCount.put(e.getKey(), e.getValue().size());
+    }
+    assertEquals(expected, resultCount);
+  }
+
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/RelMetadataTest.java]: Shall we use `checkNodeTypeCount` ?

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java [hunk 2]
```diff
@@ -2360,6 +2361,27 @@ private void checkNodeTypeCount(String sql, Map<Class<? extends RelNode>, Intege
     checkNodeTypeCount(sql, expected);
   }
 
+  @Test public void testNodeTypeExchange() {
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/RelMetadataTest.java]: Other tests for node type checking is by `testNodeTypeXXX `, shall we follow the convention ?

File: core/src/main/java/org/apache/calcite/rel/metadata/RelMdNodeTypes.java [hunk 3]
```diff
@@ -134,6 +135,11 @@
     return getNodeTypes(rel, TableModify.class, mq);
   }
 
+  public Multimap<Class<? extends RelNode>, RelNode> getNodeTypes(Exchange rel,
+      RelMetadataQuery mq) {
+    return getNodeTypes(rel, Exchange.class, mq);
+  }
+
```
[the remark this answers is not in the document]
Comment (xy2953396112) [reply core/src/main/java/org/apache/calcite/rel/metadata/RelMdNodeTypes.java]: yeah, many operators have been added.

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java (hunk 1 above)
Comment (xy2953396112) [reply core/src/test/java/org/apache/calcite/test/RelMetadataTest.java]: no, exchange cannot be expressed by SQL string.

**Pull request closed without merging**